File: src/main/java/no/nav/syfo/util/MockUtil.java
package no.nav.syfo.util;

import static java.lang.System.getenv;
import static java.util.Arrays.asList;
import static java.util.Optional.ofNullable;
import static no.nav.syfo.util.PropertyUtil.ALTINN_TEST_WHITELIST_ORGNR;

public class MockUtil {

    public static String getOrgnummerForSendingTilAltinn(String orgnummer) {

        return ofNullable(getenv(ALTINN_TEST_WHITELIST_ORGNR))
                .map(whitelist -> asList(whitelist.split(",")))
                .map(whitelist -> whitelist.contains(orgnummer) ? orgnummer : whitelist.get(0))
                .orElse(orgnummer);
    }
}

File: src/main/java/no/nav/syfo/util/MockUtil.kt
package no.nav.syfo.util

private const val defaultOrgNumber = "910067494"
private const val vismaTestUserOrgNumberInNAV = "972674818"
private const val vismaTestUserOrgNumberInAltinn = "215918912"
private val allowedOrgnumbersToBeSentToAltinn = listOf(defaultOrgNumber, vismaTestUserOrgNumberInAltinn)

fun getOrgnummerForSendingTilAltinn(orgnumber: String): String {
    if (!isDev()) return orgnumber

    return when {
        vismaTestUserOrgNumberInNAV == orgnumber -> vismaTestUserOrgNumberInAltinn
        allowedOrgnumbersToBeSentToAltinn.contains(orgnumber) -> orgnumber
        else -> defaultOrgNumber
    }

}

private fun isDev(): Boolean = System.getenv("ENVIRONMENT") == "dev"
